sketch/GameController.ts

```diff
@@ -1,51 +1,46 @@
 
-    //Draw the gameArea
-    public drawGameArea() {
-        let gameArea = new GameArea();
-        gameArea.draw();
-    }
+  //Draw the gameArea
+  public drawGameArea() {
+    let gameArea = new GameArea();
+    gameArea.draw();
+  }
 
-    //Loop list of level objects and draw them
-    public drawLevel() {        
-        let char: number = 0;
-        
-        for(let i = 0; i < this.level.levelObjects.length; i++){
-            this.level.levelObjects[i].draw();
-            
-            if (this.level.levelObjects[i].constructor == Character) {                
-                char = i;
-            } 
-        }         
-        return this.level.levelObjects[char];   //Return the Character object    
-    }
+  //Loop list of level objects and draw them
+  public drawLevel() {
+    let spawnPoint: object = {};
 
-    //Keep track of player versus all collidable objects
-    collisionDetection(character: Character) {        
-        let collideBlocks: Array<LevelObject> = []; 
+    for (let i = 0; i < this.level.levelObjects.length; i++) {
+      this.level.levelObjects[i].draw();
 
-        //Loop through all objects and save the collideable ones in new array
-        for(let i = 0; i < this.level.levelObjects.length; i++) {
-            if (this.level.levelObjects[i].constructor == Block){
-                collideBlocks.push(this.level.levelObjects[i]);
-            }
-        }
-        
-        //Determina what collision is and what should happen when collision occur
-        for(let i = 0; i < collideBlocks.length; i++){
-            //let d = dist(character.x, character.y, collideBlocks[i].x, collideBlocks[i].y);
-            if (character.x < collideBlocks[i].x + collideBlocks[i].w &&
-                character.x + character.w > collideBlocks[i].x &&
-                character.y < collideBlocks[i].y + collideBlocks[i].h 
-                &&
-                character.y + character.h > collideBlocks[i].y) {
-                    
-                    character.y = collideBlocks[i].y - character.h;
-                    character.vy = 0;                     
-             } 
-        }
+      if (this.level.levelObjects[i].constructor === StartBlock) {
+        spawnPoint = {
+          posX: this.level.levelObjects[i].x,
+          posY: this.level.levelObjects[i].y,
+          cellUnit: this.level.levelObjects[i].w
+        };
+      }
+      if (this.level.levelObjects[i].constructor === Block && player) {
+        this.level.levelObjects[i].collide();
+      }
+      if (this.level.levelObjects[i].constructor === FinishBlock && player) {
+        this.level.levelObjects[i].collide();
+      }
     }
-}
\ No newline at end of file
+    return spawnPoint; //return the point where the character shall be placed
+  }
+
+  public spawnPlayer() {
+    spawnPoint = this.drawLevel();
+    player = new Character(
+      spawnPoint.posX,
+      spawnPoint.posY,
+      spawnPoint.cellUnit
+    );
+
+    return player;
+  }
+}
```

Comment:
returning the player here instead makes it possible for us to make collision detection in each block.